### PR TITLE
Add rocks.poopjournal.librelinkupdesktop

### PR DESCRIPTION
**Pull request by CrazyMarvin**
### Please confirm your submission meets all the criteria

- [X] Please describe your application briefly.

This is a desktop application that fetches your blood sugar from [LibreLinkUp](https://librelinkup.com/).

#### Features of LibreLinkUpDesktop:
- Show blood glucose level on your desktop in a little window
- No tracking
- Dark Mode
- Libre software
- That's it. 🩸

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [X] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

**Comment by CrazyMarvin** [comment]
bot, build rocks.poopjournal.librelinkupdesktop

File: rocks.poopjournal.librelinkupdesktop.yml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+      - cp -a dist/linux*unpacked /app/main
+      # Install app wrapper
+      - install -Dm755 -t /app/bin/ ../run.sh
+    subdir: main
+    sources:
+      #- type: archive
+      #  url: https://github.com/flathub/electron-sample-app/archive/1.0.1.tar.gz
+      #  sha256: a2feb3f1cf002a2e4e8900f718cc5c54db4ad174e48bfcfbddcd588c7b716d5b
+      - type: dir
+        path: ..
```
Comment (hfiguiere) [inline rocks.poopjournal.librelinkupdesktop.yml:46]: and where do you think `..` points to?

File: rocks.poopjournal.librelinkupdesktop.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        dest: main
+      - generated-sources.json
```
Comment (hfiguiere) [inline rocks.poopjournal.librelinkupdesktop.yml:48]: and what about this?

**Review by hfiguiere — CHANGES_REQUESTED** [review]
also I don't see any icons, .desktop or appstream file installed.

**Comment by CrazyMarvin** [comment]
> also I don't see any icons, .desktop or appstream file installed.

I just copied it from the [sample](https://github.com/flathub/electron-sample-app/blob/master/flatpak/org.flathub.electron-sample-app.yml) and hoped it would build. 😅
I will have a closer look later.

**Comment by CrazyMarvin** [comment]
Do you recommend to put some files like the Flathub Manifest or MetaInfo File into the upstream repo or is it fine to have everything just here?

**Comment by hfiguiere** [comment]
> Do you recommend to put some files like the Flathub Manifest or MetaInfo File into the upstream repo or is it fine to have everything just here?

Upstream is always best. The ideal flathub repo contain only the manifest file.

**Comment by hfiguiere** [comment]
Definitely NEVER the upstream source or distribution

In that case the `generated-sources.json` is missing should be in the repo.

File: rocks.poopjournal.librelinkupdesktop.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    buildsystem: simple
+    build-options:
+      env:
+        XDG_CACHE_HOME: /run/build/librelinkupdesktop/flatpak-node/cache
+        npm_config_cache: /run/build/librelinkupdesktop/flatpak-node/npm-cache
+        npm_config_offline: 'true'
+    build-commands:
+      - npm install --offline
+      - npx electron-packager . librelinkupdesktop --overwrite --platform=linux --arch=x64 --out=release-build --prune=true --electron-version=27.3.4 --app-version=0.1.3
+      - ls -la release-build/librelinkupdesktop-linux-x64
```
Comment (hfiguiere) [inline rocks.poopjournal.librelinkupdesktop.yml:31]:
```suggestion
```

File: rocks.poopjournal.librelinkupdesktop.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+  - name: librelinkupdesktop
+    buildsystem: simple
+    build-options:
+      env:
+        XDG_CACHE_HOME: /run/build/librelinkupdesktop/flatpak-node/cache
+        npm_config_cache: /run/build/librelinkupdesktop/flatpak-node/npm-cache
+        npm_config_offline: 'true'
+    build-commands:
+      - npm install --offline
+      - npx electron-packager . librelinkupdesktop --overwrite --platform=linux --arch=x64 --out=release-build --prune=true --electron-version=27.3.4 --app-version=0.1.3
```
Comment (hfiguiere) [inline rocks.poopjournal.librelinkupdesktop.yml:30]: so how does that work on aarch64?

Comment (hfiguiere) [reply rocks.poopjournal.librelinkupdesktop.yml:30]: also app-version 0.1.3 but tarball is 0.1.4. Am I missing something?

**Comment by fperera123** [comment]
bot, build rocks.poopjournal.librelinkupdesktop

**Comment by fperera123** [comment]
> also I don't see any icons, .desktop or appstream file installed.

These are now added

File: rocks.poopjournal.librelinkupdesktop.yml (hunk 4 above)
Comment (hfiguiere) [reply rocks.poopjournal.librelinkupdesktop.yml:30]: 0.1.5

**Comment by fperera123** [comment]
bot, build rocks.poopjournal.librelinkupdesktop

File: rocks.poopjournal.librelinkupdesktop.yml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    build-commands:
+      - npm install --offline
+      - ./flatpak-node/electron-builder-arch-args.sh
+      - npx ts-node ./.erb/scripts/clean.js dist && npm run build && npx electron-builder --linux --dir $ELECTRON_BUILDER_ARCH_ARGS
+      - ls -la release/build/linux*unpacked
+      - cp -a release/build/linux*unpacked /app/main
+      - install -Dm755 -t /app/bin/ ./run.sh
+      - install -Dm644 assets/logo.svg /app/share/icons/hicolor/scalable/apps/rocks.poopjournal.librelinkupdesktop.svg
+      - install -Dm644 flathub/rocks.poopjournal.librelinkupdesktop.desktop /app/share/applications/rocks.poopjournal.librelinkupdesktop.desktop
+      - install -Dm444 flathub/rocks.poopjournal.librelinkupdesktop.metainfo.xml /app/share/metainfo/rocks.poopjournal.librelinkupdesktop.metainfo.xml
```
Comment (bbhtt) [inline rocks.poopjournal.librelinkupdesktop.yml:37]: It's `Apache-2.0` https://spdx.org/licenses/

**Comment by bbhtt** [comment]
bot, build rocks.poopjournal.librelinkupdesktop

**Comment by fperera123** [comment]
bot, build rocks.poopjournal.librelinkupdesktop

**Comment by fperera123** [comment]
bot, build rocks.poopjournal.librelinkupdesktop

**Comment by fperera123** [comment]
bot, build rocks.poopjournal.librelinkupdesktop

**Comment by fperera123** [comment]
bot, build rocks.poopjournal.librelinkupdesktop

**Comment by fperera123** [comment]
bot, build rocks.poopjournal.librelinkupdesktop

**Comment by CrazyMarvin** [comment]
> Build [112598 successful](https://buildbot.flathub.org/#/builders/6/builds/112598) To test this build, install it from the testing repository:
> 
> ```
> flatpak install --user https://dl.flathub.org/build-repo/95408/rocks.poopjournal.librelinkupdesktop.flatpakref
> ```

This looks fine. 🥳

**Comment by fperera123** [comment]
bot, build rocks.poopjournal.librelinkupdesktop

**Comment by fperera123** [comment]
bot, build rocks.poopjournal.librelinkupdesktop

File: rocks.poopjournal.librelinkupdesktop.yml [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+    build-options:
+      env:
+        XDG_CACHE_HOME: /run/build/librelinkupdesktop/flatpak-node/cache
+        npm_config_cache: /run/build/librelinkupdesktop/flatpak-node/npm-cache
+        npm_config_offline: "true"
+    build-commands:
+      - npm install --offline
+      - ./flatpak-node/electron-builder-arch-args.sh
+      - npx ts-node ./.erb/scripts/clean.js dist && npm run build && npx electron-builder --linux --dir $ELECTRON_BUILDER_ARCH_ARGS
+      - ls -la release/build/linux*unpacked
```
Comment (bbhtt) [inline rocks.poopjournal.librelinkupdesktop.yml:32]:
```suggestion
```

**Comment by fperera123** [comment]
bot, build rocks.poopjournal.librelinkupdesktop

**Comment by CrazyMarvin** [comment]
Thanks to everyone! 😍

**Pull request closed without merging**